### PR TITLE
Add cursor scaling

### PR DESCRIPTION
**Pull request by ccoors**
I'm not happy with the placement of the scaleDataArray function, but apparently there is no place where we actually do pixel operations... ? Unfortunately, SDL does not provide functions for cursor scaling.

I'm open to suggestions.

### CODE REVIEW

**Comment by mrtnptrs** [comment]
Uhhhh, not trying to be rude, but I see no difference in the size of the cursor using the build from this PR. There seems to be no difference with the latest development build. Could you maybe explain further what the effects of your changes are?

**Comment by ccoors** [comment]
Ah. Oh, okay, sorry. So in the options you can adjust the window scale factor. I have it set to 2.0 because I have a 4k monitor. Everything scaled perfectly except for the cursors, because they are not directly drawn by the game.

In this PR they are scaled with the window scale factor.

So window scale factor 0.5-1.5 means 1x cursors, 1.5-2.5 means 2x cursors, 2.5-3.5x means 3x cursors and so on.

**Comment by mrtnptrs** [comment]
Could you maybe provide a few images with some examples? As I totally don't see any difference..... but it could just be me.

**Comment by ccoors** [comment]
Sure. This is how it looks right now. See the tiny cursor on the yellow arrow:
screenshot_20171209_231407

This is how it looks with a scaled up cursor:
screenshot_20171209_231443

The 'normal' cursor and the hand cursor can't be scaled up, because they are provided by the system, but the custom cursors (like that crosshair, the path cursor, the wall cursor etc.) can.

File: src/openrct2-ui/CursorRepository.cpp [hunk 1]
```diff
@@ -14,41 +14,27 @@
 *****************************************************************************/
 #pragma endregion
 
+#include <cmath>
+
 #include <openrct2/common.h>
-#include <SDL.h>
```
Comment (Gymnasiast) [inline src/openrct2-ui/CursorRepository.cpp]: Why did you remove this include?

Comment (ccoors) [reply src/openrct2-ui/CursorRepository.cpp]: I moved it to `CursorRepository.h` because I need the SDL_Cursor definition there.

**Comment by Gymnasiast** [comment]
It looks reasonable to me.

File: src/openrct2-ui/windows/Options.cpp [hunk 2]
```diff
@@ -970,13 +970,15 @@ static void window_options_mousedown(rct_window *w, rct_widgetindex widgetIndex,
             config_save_default();
             gfx_invalidate_screen();
             context_trigger_resize();
+            context_update_cursor_scale();
```
Comment (janisozaur) [inline src/openrct2-ui/windows/Options.cpp]: there is one more place where it needs to get called – in handler for `set window_scale`

File: src/openrct2-ui/CursorRepository.cpp [hunk 3]
```diff
@@ -61,20 +47,104 @@ void CursorRepository::SetCurrentCursor(CURSOR_ID cursorId)
...
+        {
+            switch (cursorId)
+            {
+                // We can't scale the system cursors, but they should be appropriately scaled anyway
+                case CURSOR_ARROW:
+                    return SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_ARROW);
+                case CURSOR_HAND_POINT:
+                    return SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_HAND);
+                default:
+                    return Create(GetCursorData(cursorId), scale);
```
Comment (janisozaur) [inline src/openrct2-ui/CursorRepository.cpp]: as it is not immediately obvious why `this` needs to be captured, maybe write `this->Create`?
